### PR TITLE
CookieClearingLogoutHandler enhancement

### PR DESCRIPTION
**Pull request by maxcoinage**
Enabled the ability to pass in an array of Cookies to support clearing cookies on a different path other than the default context path
Issue: gh-6078

Issue #6078 

cc: @rwinch

### CODE REVIEW

**Comment by maxcoinage** [comment]
FYI - I wasn't sure what to put in for the `since` [annotation](https://github.com/spring-projects/spring-security/pull/6116/files#diff-c73a731bdd8f8ffb9435811e364771d7R44) - so put `5.X` - let me know if you want to update this

File: web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java [hunk 1]
```diff
@@ -24,28 +24,49 @@
...
+ * - A defined list of cookie names, using the context path as the cookie path
+ * OR
+ * - A given list of Cookies
  *
  * @author Luke Taylor
  * @since 3.1
  */
 public final class CookieClearingLogoutHandler implements LogoutHandler {
-	private final List<String> cookiesToClear;
+	private final List<Object> cookiesToClear;
```
Comment (rwinch) [inline web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java]: Let's change this to `List<Function<HttpServletRequest,Cookie>>`. In each of the constructors we can convert the arguments into a `Function<HttpServletRequest,Cookie>`.

File: web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java [hunk 2]
```diff
@@ -24,28 +24,49 @@
...
 
 	public CookieClearingLogoutHandler(String... cookiesToClear) {
 		Assert.notNull(cookiesToClear, "List of cookies cannot be null");
-		this.cookiesToClear = Arrays.asList(cookiesToClear);
+		this.cookiesToClear =  Arrays.asList((Object[]) cookiesToClear);
+	}
+
+	/**
+	 * @since 5.X
```
Comment (rwinch) [inline web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java]: Please change this to 5.2

File: web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java [hunk 3]
```diff
@@ -24,28 +24,49 @@
...
 			Authentication authentication) {
-		for (String cookieName : cookiesToClear) {
-			Cookie cookie = new Cookie(cookieName, null);
-			String cookiePath = request.getContextPath() + "/";
-			cookie.setPath(cookiePath);
-			cookie.setMaxAge(0);
-			response.addCookie(cookie);
+		for (Object cookie : cookiesToClear) {
+			Cookie realCookie = null;
+			if (cookie instanceof String) {
```
Comment (rwinch) [inline web/src/main/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandler.java]: This logic will obviously need updated once we change `this.cookiesToClear`.

**Comment by maxcoinage** [comment]
@rwinch  Thanks for the feedback, I was not too happy with my initial solution and think your proposal is much better  PR is now updated, please check at your convenience

**Pull request merged**